migration-guide.md

```diff
@@ -0,0 +1,500 @@
+
+```java
+CosmosContainer = client.getDatabase("MyDatabaseName").getContainer("MyContainerName");
+```
+
+In version 2.x.x Java SDKs, all operations on resources and documents are performed through the client instance.
+
+### Representing documents
+
+In Java SDK 4.0, custom POJO's and ```JsonNodes``` are the two options for writing and reading documents from Azure Cosmos DB. 
+
+In Java SDK 3.x.x ```CosmosItemProperties``` was exposed by the public API and served as a document representation. This class is no longer exposed in Java SDK 4.0.
+
+### Imports
+
+* Java SDK 4.0 packages begin with ```com.azure.cosmos```
+    * Java SDK 3.x.x packages begin with ```com.azure.data.cosmos```
+
+* Java SDK 4.0 places a number of classes in a nested package, ```com.azure.cosmos.models```. This includes
+    * ```CosmosContainerResponse```'
+    * ```CosmosDatabaseResponse```
+    * ```CosmosItemResponse```
+    * And Async API analogs of all of the above...
+    * ```CosmosContainerProperties```
+    * ```FeedOptions```
+    * ```PartitionKey```
+    * ```IndexingPolicy```
+    * ```IndexingMode```    
+    * ...etc.
+
+### Accessors
+             
+Java SDK 4.0 exposes ```get``` and ```set``` methods for accessing instance members.
+* Example: a ```CosmosContainer``` instance has ```container.getId()``` and ```container.setId()``` methods.
+
+This is different from Java SDK 3.x.x which exposes a fluent interface.
+              Example: a ```CosmosSyncContainer``` instance has ```container.id()``` which is overloaded to get or set ```id```.
+
+## Code snippet comparisons
+
+### Create resources
+
+**Java SDK 4.0 Async API:**
+
+```java
+ConnectionPolicy defaultPolicy = ConnectionPolicy.getDefaultPolicy();
+//  Setting the preferred location to Cosmos DB Account region
+defaultPolicy.setPreferredLocations(Lists.newArrayList("Your Account Location"));
+// Use Direct Mode for best performance
+defaultPolicy.setConnectionMode(ConnectionMode.DIRECT);
+
+// Create Async client.
+// Building an async client is still a sync operation.
+client = new CosmosClientBuilder()
+        .setEndpoint("your.hostname")
+        .setKey("yourmasterkey")
+        .setConnectionPolicy(ConnectionPolicy.getDefaultPolicy())
+        .setConsistencyLevel(ConsistencyLevel.EVENTUAL)
+        .buildAsyncClient();
+
+
+// Create database with specified name
+client.createDatabaseIfNotExists("YourDatabaseName")
+      .flatMap(databas'Response -> {
+        database = databaseResponse.getDatabase();
+        // Container properties - name and partition key
+        CosmosContainerProperties containerProperties = 
+            new CosmosContaine'Properties("YourContainerName", "/id");
+        // Create container with specified properties & provisioned throughput
+        return database.createContainerIfNotExists(containerProperties, 400);
+    }).flatMap(containerResponse -> {
+        container = containerResponse.getContainer();
+        return Mono.empty();
+}).subscribe();
```

Comment:
in some places in doc you used block() and some places subscribe()
I would recommend we don't use subscribe() in general. as this is a simple guide, let's use block()